lib/membrane_h264_plugin/parser.ex

```diff
@@ -276,34 +435,109 @@ defmodule Membrane.H264.Parser do
     end)
   end
 
-  defp prepare_actions_for_aus(aus, state) do
+  @spec prepare_actions_for_aus(
+          [AUSplitter.access_unit()],
+          CallbackContext.t(),
+          state()
+        ) :: callback_return()
+  defp prepare_actions_for_aus(aus, ctx, state) do
     Enum.flat_map_reduce(aus, state, fn au, state ->
-      {sps_actions, state} = maybe_parse_sps(au, state)
-
-      au = maybe_add_parameter_sets(au, state) |> delete_duplicate_parameter_sets()
-      state = cache_parameter_sets(state, au)
+      {au, stream_format_actions, state} = process_au_parameter_sets(au, ctx, state)
 
       {{pts, dts}, state} = prepare_timestamps(au, state)
 
       buffers_actions = [
-        {:buffer, {:output, wrap_into_buffer(au, pts, dts, state.output_alignment)}}
+        buffer:
+          {:output,
+           wrap_into_buffer(au, pts, dts, state.output_alignment, state.output_stream_structure)}
       ]
 
-      {sps_actions ++ buffers_actions, state}
+      {stream_format_actions ++ buffers_actions, state}
     end)
   end
 
-  defp maybe_parse_sps(au, state) do
-    case Enum.find(au, &(&1.type == :sps)) do
-      nil ->
-        {[], state}
+  @spec process_new_parameter_sets([NALu.t()], [NALu.t()], CallbackContext.t(), state()) ::
+          {[Action.t()], state()}
+  defp process_new_parameter_sets(new_spss, new_ppss, context, state) do
+    updated_cached_spss = merge_parameter_sets(new_spss, state.cached_spss, :seq_parameter_set_id)
+    updated_cached_ppss = merge_parameter_sets(new_ppss, state.cached_ppss, :pic_parameter_set_id)
+
+    state = %{state | cached_spss: updated_cached_spss, cached_ppss: updated_cached_ppss}
+
+    latest_sps = List.last(new_spss)
+
+    last_sent_stream_format = context.pads.output.stream_format
+
+    output_raw_stream_structure =
+      case state.output_stream_structure do
+        :annexb ->
+          :annexb
+
+        {avc, _nalu_length_size} = output_stream_structure ->
```

Comment:
let's write our own guard: `is_avc(output_stream_structure)`

Reply:
in this case we are still using the extracted avc atom. I think there are no cases where we are only checking if the stream format is avc and not extracting anything